pkg/graph/resolvers/itgovactions/newstep/progress_to_new_step.go

```diff
@@ -0,0 +1,223 @@
+			ActionType:     models.ActionTypePROGRESSTONEWSTEP,
+			ActorName:      adminUserInfo.CommonName,
+			ActorEmail:     adminUserInfo.Email,
+			ActorEUAUserID: adminEUAID,
+			Step:           &stepForAction,
+		}
+		if input.AdditionalNote != nil {
+			action.Feedback = null.StringFromPtr(input.AdditionalNote)
+		}
+
+		_, errCreatingAction = store.CreateAction(ctx, &action)
+		if errCreatingAction != nil {
+			return errCreatingAction
+		}
+
+		return nil
+	})
+
+	// save feedback for requester
+	if input.Feedback != nil {
+		errGroup.Go(func() error {
+			feedbackForRequester := &models.GovernanceRequestFeedback{
+				IntakeID:     intake.ID,
+				Feedback:     *input.Feedback,
+				SourceAction: models.GRFSAProgressToNewStep,
+				TargetForm:   models.GRFTFNoTargetProvided,
+				Type:         models.GRFTRequester,
+			}
+			feedbackForRequester.CreatedBy = adminEUAID
+
+			_, errRequesterFeedback := store.CreateGovernanceRequestFeedback(ctx, feedbackForRequester)
+			if errRequesterFeedback != nil {
+				return errRequesterFeedback
+			}
+
+			return nil
+		})
+	}
+
+	// save feedback/recommendations for GRB
+	if input.GrbRecommendations != nil {
+		errGroup.Go(func() error {
+			feedbackForGRB := &models.GovernanceRequestFeedback{
+				IntakeID:     intake.ID,
+				Feedback:     *input.GrbRecommendations,
+				SourceAction: models.GRFSAProgressToNewStep,
+				TargetForm:   models.GRFTFNoTargetProvided,
+				Type:         models.GRFTGRB,
+			}
+			feedbackForGRB.CreatedBy = adminEUAID
+
+			_, errGRBFeedback := store.CreateGovernanceRequestFeedback(ctx, feedbackForGRB)
+			if errGRBFeedback != nil {
+				return errGRBFeedback
+			}
+
+			return nil
+		})
+	}
+
+	// save admin note
+	if input.AdminNote != nil {
+		errGroup.Go(func() error {
+			adminUserInfo, errAdminUserInfo := fetchUserInfo(ctx, adminEUAID)
+			if errAdminUserInfo != nil {
+				return errAdminUserInfo
+			}
+
+			adminNote := &models.SystemIntakeNote{
+				SystemIntakeID: input.SystemIntakeID,
+				AuthorEUAID:    adminEUAID,
+				AuthorName:     null.StringFrom(adminUserInfo.CommonName),
+				Content:        null.StringFromPtr(input.AdminNote),
+			}
+
+			_, errCreateNote := store.CreateSystemIntakeNote(ctx, adminNote)
+			if errCreateNote != nil {
+				return errCreateNote
+			}
+
+			return nil
+		})
+	}
+
+	if err := errGroup.Wait(); err != nil {
+		return nil, err
+	}
+
+	return updatedIntake, nil
+}
+
+// the frontend shouldn't allow users to take any invalid actions, but validate server-side to make sure
+func isIntakeValid(intake *models.SystemIntake, newStep model.SystemIntakeStepToProgressTo) error {
+	if intake.State == models.SystemIntakeStateCLOSED {
+		return &apperrors.InvalidActionError{
+			ActionType: models.ActionTypePROGRESSTONEWSTEP,
+			Message:    "Can't take Progress to New Step action on closed intakes",
+		}
+	}
+
+	if intake.Step == models.SystemIntakeStepINITIALFORM && intake.RequestFormState == models.SIRFSNotStarted {
```

Comment:
Is this constraint requested for by UX? It seems to me the general trend is allowing flexibility in the workflow.  (Though I do see the logic of checking this)

Reply:
It wasn't explicitly requested, but an intake that hasn't been started isn't visible to admins in the UI, so an action on an intake in that state shouldn't be allowed.